src/EnergyPlus/CommandLineInterface.cc

```diff
@@ -322,143 +331,168 @@ namespace CommandLineInterface {
+        DataStringGlobals::outputTblCsvFilePath = composePath(tableSuffix + ".csv");
+        DataStringGlobals::outputTblHtmFilePath = composePath(tableSuffix + ".htm");
+        DataStringGlobals::outputTblTabFilePath = composePath(tableSuffix + ".tab");
+        DataStringGlobals::outputTblTxtFilePath = composePath(tableSuffix + ".txt");
+        DataStringGlobals::outputTblXmlFilePath = composePath(tableSuffix + ".xml");
+        state.files.outputMapTabFilePath = composePath(mapSuffix + ".tab");
+        state.files.outputMapCsvFilePath = composePath(mapSuffix + ".csv");
+        state.files.outputMapTxtFilePath = composePath(mapSuffix + ".txt");
+        state.files.outputZszCsvFilePath = composePath(zszSuffix + ".csv");
+        state.files.outputZszTabFilePath = composePath(zszSuffix + ".tab");
+        state.files.outputZszTxtFilePath = composePath(zszSuffix + ".txt");
+        state.files.outputSszCsvFilePath = composePath(sszSuffix + ".csv");
+        state.files.outputSszTabFilePath = composePath(sszSuffix + ".tab");
+        state.files.outputSszTxtFilePath = composePath(sszSuffix + ".txt");
+        DataStringGlobals::outputAdsFilePath = composePath(adsSuffix + ".out");
+        state.files.shade.filePath = composePath(shdSuffix + ".csv");
+
+        if (suffixType == "L") {
+            // out/sqlite.er:w
+            DataStringGlobals::outputSqliteErrFilePath = fs::path(DataStringGlobals::outDirPath.string() + sqliteSuffix + ".err");
         } else {
-            outputSqliteErrFileName = outputFilePrefix + sqliteSuffix + ".err";
+            // if 'D':  out/eplus-sqlite.err
+            DataStringGlobals::outputSqliteErrFilePath = composePath(sqliteSuffix + ".err");
         }
-        state.files.screenCsv.fileName = outputFilePrefix + screenSuffix + ".csv";
-        state.files.delightIn.fileName = "eplusout.delightin";
-        outputDelightOutFileName = "eplusout.delightout";
-        state.files.iniFile.fileName = "Energy+.ini";
-        state.files.inStatFileName.fileName = weatherFilePathWithoutExtension + ".stat";
-        eplusADSFileName = inputDirPathName + "eplusADS.inp";
+
+        state.files.screenCsv.filePath = composePath(screenSuffix + ".csv");
+        // TODO, why are these relative paths?
+        state.files.delightIn.filePath = "eplusout.delightin";
+        DataStringGlobals::outputDelightOutFilePath = "eplusout.delightout";
+
+        // TODO: why is this relative?
+        state.files.iniFile.filePath = "Energy+.ini";
+
+        // Stat next to epw
+        // Careful! fs::path::replace_extension will **mutate the original object**
+        state.files.inStatFilePath.filePath = state.files.inputWeatherFilePath.filePath;
+        state.files.inStatFilePath.filePath.replace_extension(".stat");
+        // Or is it better to provide a helper like this?
+        // state.files.inStatFilePath.filePath = FileSystem::replaceFileExtension(state.inputWeatherFilePath.filePath, ".stat");
```

Comment:
Are you saying you could accidentally stomp on a previous value because it is mutating the underlying reference?  I don't think a helper should be really necessary, primarily because if a developer messes with that the wrong way, the simulation and the CI system should complain with huge red badges alerting that things went terribly wrong.  But, all that said, if a worker function can help avoid that ever happening, then OK, maybe it's worth it.

Reply:
It mutates the path it's called on yes

See an example on https://godbolt.org/z/Mvze3W

```
#include <filesystem>
#include <iostream>

namespace fs = std::filesystem;

int main() {

    fs::path inputWeatherFilePath("/path/to/myfile.epw");
    fs::path inStatFilePath = inputWeatherFilePath.replace_extension(".stat");
    std::cout << "inputWeatherFilePath=" << inputWeatherFilePath.string() << '\n'; // !! => inputWeatherFilePath.filePath
    std::cout << "inStatFilePath=" << inStatFilePath.string() << '\n';

}
```

The thing with having a helper function is that nothing prevents anyone from **not** using it and directly calling the `fs::path::replace_extension` instead of the helper...